### DISCUSSION TITLE
convert to lowercase before using beginswith operator

### DISCUSSION THREAD

**Discussion by ag0059985** [react-querybuilder/react-querybuilder#799] [Q&A] [answered]
i am relatively new to react-querybuilder, just needed help in using lowercase function before comparing value with beginswith operation. if anyone could pls share the codesandbox link for it.
thanks in advance

**Comment by jakeboone02** [comment, marked as answer]
I'm going to assume you're talking about the export feature, `formatQuery`, since the `QueryBuilder` component itself doesn't manipulate the values like that.

If you look at the bottom of the [Rule processor](https://react-querybuilder.js.org/docs/utils/export#rule-processor) section of the [Export](https://react-querybuilder.js.org/docs/utils/export) documentation, you'll see this example:

```ts
const customRuleProcessor: RuleProcessor = (rule, options) => {
  if (rule.operator === 'has') {
    return `UPPER(${rule.field}) LIKE UPPER('%${rule.value}%')`;
  }

  return defaultRuleProcessorSQL(rule, options);
};

formatQuery(query, { format: 'sql', ruleProcessor: customRuleProcessor });
/*
"(UPPER(firstName) LIKE UPPER('%S%') and lastName = 'Vai')"
 ^------------custom--------------^ ^------default-----^
*/
```

You can modify that rule processor to key on `rule.operator === 'beginsWith'` and use `LOWER` (or the corresponding function for your environment) instead of `UPPER`.

**Comment by ag0059985** [comment]
thanks so much for the response @jakeboone02
